### ISSUE TITLE
[FEA] Support local_dict keyword in query

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#1322]
**Is your feature request related to a problem? Please describe.**
As a user, I'd like to use the `query` API with dask for scaling out queries containing local 
variables. Because cuDF searches up the stack to grab the variable, it's currently not passed through to dask. In order to effectively use the `query` API with dask's existing implementation, we'd want to allow local variables to be used via direct passing in addition to the `@` functionality.

**Describe the solution you'd like**
I'd like to be able to do

```
df = cudf.DataFrame([('a', list(range(20))),
('b', list(reversed(range(20)))),
('c', list(range(20)))])

df.query("a == @val", local_dict={'val':10})
```

And get the same results as:

```
df = cudf.DataFrame([('a', list(range(20))),
('b', list(reversed(range(20)))),
('c', list(range(20)))])

val = 10
print(df.query("a == @val"))
    a  b   c
10  10  9  10
```

**Describe alternatives you've considered**
Numeric fields are able to be worked around by directly writing the value into the query string. Currently, there is no effective work around for some data types. Datetimes, for example, appear to require being passed as local variables to avoid running into type mismatch errors.

**Additional context**
This is relevant to the discussion here: https://github.com/rapidsai/dask-cudf/issues/128

**Comment by beckernick** [comment]
Thanks for picking this up @AK-ayush . In a branch, I tested your PR combined with the suggested dask-cudf changes in the dask-cudf issue referenced above. This will allow us to use the query API in dask with cudf dataframes. @mrocklin , I'll make a PR for this on dask-cudf this afternoon.

**Comment by kkraus14** [comment]
Fixed by #1335
